corehq/apps/reports/standard/cases/data_sources.py

```diff
@@ -230,43 +159,241 @@ def opened_by_user_id(self):
         else:
             return user['id']
 
+    def __getattr__(self, item):
+        if item in self.aliases:
+            item = self.aliases[item]
+        return object.__getattribute__(self, item)
```

Comment:
[`__getattr__`](https://docs.python.org/3/reference/datamodel.html#object.__getattr__) is only called if the attribute does not otherwise exist on the object, so calling `__getattribute__` is pointless, it will always raise.
```suggestion
        raise AttributeError(item)
```

Reply:
Thanks, this still isn't quite right since I still need to call `getattr` with the alias name but the specific line should be an exception: d340b2af882

I was having some difficulty here that I couldn't quite get my head around. It seems that if a class property raises an `AttributeError` then python will call `__getattr__` even though the class has a property with that name. This was causing some recursion issues when I was always calling `getattr`.

You can see what I mean here: https://replit.com/@snopoke/Python-getattr-with-class-properties?v=1#main.py

Reply:
> Thanks, this still isn't quite right since I still need to call `getattr` with the alias name...

Oh, this was originally written differently. I had forgotten that changed in https://github.com/dimagi/commcare-hq/pull/31779/commits/bed33e5fb8cae47ee296fa06cc04669c38cdac85